src/utilities/graphqlConnectionFactory.ts

```diff
@@ -1,38 +1,178 @@
+export const getSortingObject = (
+  direction: "FORWARD" | "BACKWARD",
+  sortingObject: { [key: string]: number }
+): { [key: string]: number } => {
+  // We assume that the resolver would always be written with respect to the sorting that needs to be implemented for forward pagination
+  if (direction === "FORWARD") return sortingObject;
+
+  // If we are paginating backwards, then we must reverse the order of all fields that are being sorted by.
+  for (const [key, value] of Object.entries(sortingObject)) {
+    sortingObject[key] = value * -1;
+  }
+
+  return sortingObject;
+};
+
+type FilterObjectType = {
+  _id: {
+    [key: string]: string;
+  };
+};
+
+// Generates the sorting arguments for filterQuery that can be passed into the .find() method
+export function getFilterObject(
+  args: CursorPaginationInput
+): FilterObjectType | null {
+  if (args.cursor) {
+    if (args.direction === "FORWARD") return { _id: { $gte: args.cursor } };
+    else return { _id: { $lte: args.cursor } };
+  }
+
+  return null;
+}
+
+/*
+This is a function which generates a GraphQL cursor pagination object based on the requirements of the client.
+
+The function takes the following arguments: 
+
+A. TYPE PARAMETERS
+
+1. T1: Refers the type of the node that the connection and it's edges would refer. 
+Example values include `Interface_User`, `Interface_Organization`, `Interface_Event`, `Interface_Post`.
+2. T2: Regers to the type of interface that is implemented by the model that you want to query. 
+For example, if you want to query the TagUser Model, then you would send Interface_UserTag for this type parameter
+
+B. DATA PARAMETERS
+
+1. args: These are the tranformed arguments that were orginally passed in the request.
+2. allFetchedObjects: Refers to objects that were fetched from the database thorugh a query.
+3. getNodeFromResult: Describes a transformation function that given an object of type T2, would convert it to the desired object of type T1. This would mostly include mapping to some specific field of the fetched object.
+
+The function returns a promise which would resolve to the desired connection object (of the type InterfaceConnection<T1>).
+*/
+export function generateConnectionObject<
+  T1 extends { _id: Types.ObjectId },
+  T2 extends { _id: Types.ObjectId }
+>(
+  args: CursorPaginationInput,
+  allFetchedObjects: T2[] | null,
+  getNodeFromResult: GetNodeFromResultFnType<T1, T2>
+): InterfaceConnectionResult<T1> {
+  // Initialize the connection object
+  const connectionObject = graphqlConnectionFactory<T1>();
+
+  // Return the default object if the recieved list is empty
+  if (!allFetchedObjects || allFetchedObjects.length === 0)
+    return {
+      data: connectionObject,
+      errors: [],
+    };
+
+  // Handling the case when the cursor is provided
+  if (args.cursor) {
+    // Validate the cursor
+    if (allFetchedObjects[0]._id.toString() !== args.cursor) {
```

Comment:
Is this a realistic check? If the cursor is provided and it is invalid, you either get a null or an empty list back which have already been tested against. If there's even one item in the list, that means cursor should be valid right?

Reply:
Not necessarily. Since the check of the cursor is of `>=` or `<=` type, if I supply any cursor less than or greater than the valid cursor in both of the cases, respectively, the fetched results would be the same. A very simple cursor of `0` will be enough to hack the system, so to speak (though why any client may do so it something that's beyond me).

Reply:
Wait i didn't get that, what do you mean `0` formatted as mongoose object id and how does it relate to hacking the system?

Reply:
Just simply meant that if we send 0 as a cursor in this implementation, it would be equivalent to sending to no cursor and all the results would be fetched.
Similary if lets say `abcd1` is a valid cursor, even `abcd2` would be treated as a valid cursor without this check (even though `abcd2` would have no object corresponding to it in the database)

Reply:
Since $gte is greater than equal, it won't pass if the objectId is not valid right?

Reply:
It will. Lets say the objects are `abcd1`, `abcd3` and `abcd5`.
Then `abcd1` is a valid cursor, and would fetch all three of these objects, and then subsequently drop the first object as per this implementation and check.
If we are to remove it, then sending `abcd2` would also fetch the objects `abcd3` and `abcd5`, even though it isn't valid.

Reply:
That would be a waste of a database request in my opnion. Why make a separate database request when we can validate the same in one?

Reply:
@EshaanAgg I'm not much aware with the advanced syntax of mongodb and mongoose, it could be that we can do both steps in a single query. I don't know. But yeah we have to use two queries in each of those resolvers. Again, the redundancy of doing the same thing can probably be abstracted away in a function or something.

Reply:
@xoldyckk Got it. I have implemented the same in each resolver! PTAL.